dspace-api/src/main/java/org/dspace/app/canvasdimension/CanvasDimensionCLI.java

```diff
@@ -0,0 +1,202 @@
+        } else {
+            System.out.println("An identifier for a Community, Collection, or Item must be provided.");
+            System.exit(1);
+        }
+        if (line.hasOption('m')) {
+            max2Process = Integer.parseInt(line.getOptionValue('m'));
+            if (max2Process <= 1) {
+                System.out.println("Invalid maximum value '" +
+                    line.getOptionValue('m') + "' - ignoring");
+                max2Process = Integer.MAX_VALUE;
+            }
+        }
+        String[] skipIds;
+
+        if (line.hasOption('s')) {
+            //specified which identifiers to skip when processing
+            skipIds = line.getOptionValues('s');
+
+            if (skipIds == null || skipIds.length == 0) {   //display error, since no identifiers specified to skip
+                System.err.println("\nERROR: -s (-skip) option requires at least one identifier to SKIP.\n" +
+                    "Make sure to separate multiple identifiers with a comma!\n" +
+                    "(e.g. -s 123456789/34,123456789/323)\n");
+                HelpFormatter myhelp = new HelpFormatter();
+                myhelp.printHelp("Canvas Dimensions\n", options);
+                System.exit(1);
+            }
+            canvasProcessor.setSkipList(Arrays.asList(skipIds));
+        }
+
+
+        DSpaceObject dso = HandleServiceFactory.getInstance().getHandleService().resolveToObject(context, identifier);
+        if (dso == null) {
+            throw new IllegalArgumentException("Cannot resolve "
+                + identifier + " to a DSpace object");
+        }
+
+        EPerson user;
+
+        if (eperson == null) {
+            System.out.println("You must provide an eperson using the \"-e\" flag.");
+            System.exit(1);
+        }
+
+        if (eperson.indexOf('@') != -1) {
+            // @ sign, must be an email
+            user = epersonService.findByEmail(context, eperson);
+        } else {
+            user = epersonService.find(context, UUID.fromString(eperson));
+        }
+
+        if (user == null) {
+            System.out.println("Error, eperson cannot be found: " + eperson);
+            System.exit(1);
+        }
+
+        context.setCurrentUser(user);
+
+        canvasProcessor.setForceProcessing(force);
+        canvasProcessor.setMax2Process(max2Process);
+        canvasProcessor.setIsQuiet(isQuiet);
+
+        int processed = 0;
+        switch (dso.getType()) {
+            case Constants.COMMUNITY:
+                processed = canvasProcessor.processCommunity(context, (Community) dso);
+                break;
+            case Constants.COLLECTION:
+                processed = canvasProcessor.processCollection(context, (Collection) dso);
+                break;
+            case Constants.ITEM:
+                canvasProcessor.processItem(context, (Item) dso);
+                processed = 1;
+                break;
+            default:
+                break;
```

Comment:
We might want to log an error of some sort if an unsupported object type is passed in. Otherwise, it appears this will just "do nothing" and be quiet about it.